diagnostic.js

```diff
@@ -32,25 +32,80 @@ const Movement = require('./models/movement.js');
 
 /// ADD YOUR CODE BELOW
 
-const create = (name, description, startYear, endYear) => {};
+const create = (name, description, startYear, endYear) => {
+  Movement.create({
+    name: name,
+    description: description, // already limited by the model, not sure if extra validation required
+    startYear: startYear,
+    endYear: endYear
+  }).then(function (movement) {
+    console.log(movement);
+  }).catch(console.error);
+};
 // Success -> console.log new Movement as JSON
 // Failure -> Console.error
 
-const index = () => {};
+const index = () => {
+  // do we need to implement the search functionality here?
+  Movement.find({}).then(function (movements) {
+    movements.forEach(function (movement) {
+      console.log(movement.toJSON());
+    });
+  }).catch(console.error);
+};
 // Success -> console.log all Movements as JSON
 // Failure -> Console.error
 
-const show = (id) => {};
+const show = (id) => {
+  Movement.findById(id).then(function (movement) {
+    // I'm very unclear about what this function will return if the movement
+    // doesn't exist, I'm assuming some falsy value, probably undefined
+    // Alternatively it might error out?
+    // The mongoose docs don't mention anything about it.
+    if (movement) {
+      return console.log(movement.toObject());
+    } else {
+      return console.log('Not Found');
+    }
+  }).catch(console.error);
+};
 // Success -> If the specified Movement exists, console.log it as JSON;
 //              otherwise, console.log "Not Found" and exit.
 // Failure -> Console.error
 
-const update = (id, field, value) => {};
+const update = (id, field, value) => {
+  let modify = {};
+  modify[field] = value;
+  Movement.findById(id)
+  .then((movement) => {
+    movement.set(field, value);
```

Comment:
The existence of the document should be checked before attempting to use it.  Otherwise, an error will be thrown.